core/src/main/java/org/apache/druid/java/util/common/parsers/ParseException.java

```diff
@@ -38,26 +40,43 @@
 public class ParseException extends RuntimeException
 {
   private final boolean fromPartiallyValidRow;
+  private final long timeOfExceptionMillis;
+  private final String input;
 
-  public ParseException(String formatText, Object... arguments)
+  public ParseException(@Nullable String input, String formatText, Object... arguments)
   {
     super(StringUtils.nonStrictFormat(formatText, arguments));
+    this.input = input;
     this.fromPartiallyValidRow = false;
+    this.timeOfExceptionMillis = System.currentTimeMillis();
   }
 
-  public ParseException(boolean fromPartiallyValidRow, String formatText, Object... arguments)
+  public ParseException(@Nullable String input, boolean fromPartiallyValidRow, String formatText, Object... arguments)
```

Comment:
nitpick - can you add a comment that input is the string representation of the input data on which the ParseException is being thrown for?

Reply:
Added a comment